state/state.go

```diff
@@ -252,7 +254,7 @@ func (state State) MakeBlock(
 		state.Version.Consensus, state.ChainID,
 		timestamp, state.LastBlockID,
 		state.Validators.Hash(), state.NextValidators.Hash(),
-		types.HashConsensusParams(state.ConsensusParams), state.AppHash, state.LastResultsHash,
```

Comment:
should we go ahead and remove `state.AppHash` and `state.LastResultsHash`?

Reply:
like previously discussed, if the createBlock is not valid, we have to use state.AppHash to record the last valid block